File: apps/android/feature/ai/data/src/main/java/com/scribblefit/feature/ai/data/repository/AuthRepositoryImpl.kt
package com.scribblefit.feature.ai.data.repository

import com.scribblefit.core.network.ScribbleFitApi
import com.scribblefit.core.network.model.AuthRequest
import com.scribblefit.feature.ai.domain.repository.AuthRepository
import com.scribblefit.feature.ai.domain.security.SecureKeyStorage
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AuthRepositoryImpl @Inject constructor(
    private val api: ScribbleFitApi,
    private val secureKeyStorage: SecureKeyStorage
) : AuthRepository {

    override suspend fun login(deviceId: String): Result<Unit> = runCatching {
        val response = api.login(AuthRequest(deviceId))
        secureKeyStorage.saveAuthToken(response.token)
    }

    override suspend fun isLogged(): Boolean {
        return secureKeyStorage.getAuthToken() != null
    }
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Data/Repository/AuthRepositoryImpl.swift
import Foundation

public final class AuthRepositoryImpl: AuthRepository {
    private let networkClient: ScribbleFitNetworkClient
    private let secureKeyStorage: SecureKeyStorage
    
    public init(networkClient: ScribbleFitNetworkClient = .shared, secureKeyStorage: SecureKeyStorage) {
        self.networkClient = networkClient
        self.secureKeyStorage = secureKeyStorage
    }
    
    public func login(deviceId: String) async throws {
        let request = AuthRequest(deviceId: deviceId)
        let response = try await networkClient.login(request: request)
        try await secureKeyStorage.saveAuthToken(response.token)
    }
    
    public func isLogged() async throws -> Bool {
        return try await secureKeyStorage.getAuthToken() != nil
    }
}
